### PR TITLE
Add a Kubernetes policy which allows the specific external IPs

### PR DESCRIPTION
**Pull request by ZeroMagic**
Ref: Azure/AKS#2012 https://github.com/Azure/Community-Policy/pull/152
Add a new Kubernetes policy to solve a CVE. 
/cc @RamyasreeChakka @EmandM

### CODE REVIEW

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        package k8sexternalips
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Service"
+          input.review.kind.group == ""
+          allowedIPs := {ip | ip := input.parameters.allowedIPs[_]}
+          externalIPs := {ip | ip := input.review.object.spec.externalIPs[_]}
+          forbiddenIPs := externalIPs - allowedIPs
+          count(forbiddenIPs) > 0
+          msg := sprintf("service has forbidden external IPs: %v", [forbiddenIPs])
```
Comment (EmandM) [inline built-in-references/Kubernetes/allowed-external-ips/template.yaml:30]: This message should read `"Service has forbidden external IPs: %v. Allowed IPs: %v"`

**Review by EmandM — CHANGES_REQUESTED** [review]

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml (hunk 1 above)
Comment (ZeroMagic) [reply built-in-references/Kubernetes/allowed-external-ips/template.yaml:30]: As we discussed here https://github.com/Azure/Community-Policy/pull/152#discussion_r543836768, currently we can keep it. If a user asks for the allowed IP to be returned, we will revisit it.

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sAzureExternalIPs
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            allowedIPs:
```
Comment (RamyasreeChakka) [inline built-in-references/Kubernetes/allowed-external-ips/template.yaml:14]: Rename 'allowedIPs' to 'allowedExternalIPs'. The parameter name here should match with that of the parameter name specified in constraint.yaml.

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sexternalips
```
Comment (RamyasreeChakka) [inline built-in-references/Kubernetes/allowed-external-ips/template.yaml:21]: Rename the package name as 'k8sazureexternalips' matching with the template name.

**Review by RamyasreeChakka — CHANGES_REQUESTED** [review]
Mismatch in the parameter names, please fix them.

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml (hunk 2 above)
Comment (ZeroMagic) [reply built-in-references/Kubernetes/allowed-external-ips/template.yaml]: done

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml (hunk 3 above)
Comment (ZeroMagic) [reply built-in-references/Kubernetes/allowed-external-ips/template.yaml]: done

File: built-in-references/Kubernetes/allowed-external-ips/template.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sazureexternalips
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Service"
+          input.review.kind.group == ""
+          allowedExternalIPs := {ip | ip := input.parameters.allowedExternalIPs[_]}
+          externalIPs := {ip | ip := input.review.object.spec.externalIPs[_]}
+          forbiddenIPs := externalIPs - allowedIPs
```
Comment (RamyasreeChakka) [inline built-in-references/Kubernetes/allowed-external-ips/template.yaml:28]: This should be allowedExternalIPs

**Review by RamyasreeChakka — APPROVED** [review]

**Review by EmandM — APPROVED** [review]

**Pull request merged**